backend/tournesol/suggestions/graph.py

```diff
@@ -0,0 +1,310 @@
+            for va in self.nodes:
+                for vb in self.nodes:
+                    if (
+                            va in scaling_factor_increasing_videos
+                            and vb in scaling_factor_increasing_videos
+                    ):
+                        va.video1_score = 1
+                        va.video2_score[vb] = 1
+                    else:
+                        va.video1_score = 0
+                        va.video2_score[vb] = 0
+        # Once the scaling factor is high enough, check what video should gain
+        # information being compared by the user
+        else:
+            sub_graphs = self.find_connex_subgraphs()
+            if len(sub_graphs) == 1:
+                sub_graphs = [self]
+            for sg in sub_graphs:
+                if len(sub_graphs) > 1:
+                    sg.compute_offline_parameters(scaling_factor_increasing_videos)
+                eigenvalues = np.linalg.eigvalsh(sg.normalized_adjacency_matrix)
+                max_beta = 0
+                for vb in sg.nodes:
+                    for va in self.nodes:
+                        # In the case the eigen value is big enough
+                        # => the graph is poorly connected,
+                        # so we should improve connexity
+                        if eigenvalues[1] > self.LAMBDA_THRESHOLD:
+                            u_index = sg.nodes.index(vb)
+                            v_index = sg.nodes.index(va)
+                            va.video2_score[vb] = sg.similarity_matrix[u_index, v_index]
+                        elif va not in sg.nodes:
+                            va.video2_score[vb] = 1
+                        else:
+                            va.video2_score[vb] = 0
+                        # Compute estimated information gain relative to the respective
+                        # uncertainties in both scores
+                        va.beta[vb] = user.delta_theta.get(vb, actual_scaling_accuracy) + \
+                            user.delta_theta.get(va, actual_scaling_accuracy) / (
+                            user.theta.get(vb, self.local_user_mean) -
+                            user.theta.get(va, self.local_user_mean) + 1
+                            )
+                        if max_beta < va.beta[vb]:
+                            max_beta = va.beta[vb]
+                for vb in sg.nodes:
+                    for va in self.nodes:
+                        va.video2_score[vb] += va.beta[vb] / max_beta
+
+    # This doesn't depend on the user -> not done here,
+    # well actually yes but not used in most of the graphs
+    def update_preferences(self):
+        """
+        Function computing online the estimated preferences with respect to another video
+        """
+        for v in self.nodes:
+            for u in set(self.graph[v]):
+                v.nb_comparison_with[u.uid] = 0
+            for u in self.graph[v]:
+                v.nb_comparison_with[u.uid] += 1
```

Comment:
still useful?

Reply:
It was to recompute all the preferences, but no more useful, that's right, thanks :)